prototype/mobile_endpoint/backends/mongo/dao.py

```diff
@@ -1,58 +1,70 @@
-from collections import defaultdict
+from uuid import UUID
+from mongoengine import DoesNotExist
 from mobile_endpoint.dao import AbsctractDao, to_generic
-from mobile_endpoint.backends.mongo.models import MongoForm, MongoCase
-from couchdbkit import ResourceNotFound
+from mobile_endpoint.backends.mongo.models import MongoForm, MongoCase, \
+    MongoSynclog
+from mobile_endpoint.exceptions import NotFound
 from mobile_endpoint.utils import get_with_lock
 
 
 class MongoDao(AbsctractDao):
 
     def commit_atomic_submission(self, xform, case_result):
-
-        docs_by_collection = defaultdict(list)
+        # TODO: Save all in one bulk operation
+        # TODO: Are the forms, cases, and synclogs all supposed to be all or nothing?
```

Comment:
So it looks like Mongo doesn't support atomic multi-document transactions. Their docs do suggest implementing a [two phase commit](http://docs.mongodb.org/manual/tutorial/perform-two-phase-commits/) pattern at the application level. But, that gives only transaction-*like* behavior. It is still possible for intermediate data to be returned at intermediate points during the two phase commit or rollback. A two phase commit also comes at the expense of performing extra writes for the transaction record and such.

So, I'm leaning towards thinking it's not worth implementing a two phase commit. Thoughts?
@czue @snopoke (I noticed the couch prototype backend as currently implemented also isn't all or nothing. Not sure if that's just because we're still iterating.)